scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -540,80 +626,82 @@ EnhancedVolcano::EnhancedVolcano(deseq_results,
-
-```{r}
+```{r expression umap}
 # filter to just myoblast cells and remove any NA's before plotting
 myoblast_combined_sce <- rms_sce[, which(rms_sce$celltype_broad == "Tumor_Myoblast")]
 
-# first look at just ARMS vs. ERMS 
+# plot PTPRT expression in ARMS vs. ERMS
 scater::plotReducedDim(myoblast_combined_sce,
                        dimred = "fastmnn_UMAP",
                        colour_by = "ENSG00000196090", #PTPRT
                        point_size= 0.5,
                        point_alpha = 0.4,
                        other_fields = "diagnosis_group") +
-  facet_wrap(~ diagnosis_group, nrow = 3)
+  facet_grid(cols = vars(diagnosis_group))
 ```
 
+In the above plot we only plotted the tumor myoblast cells that we used in our DE analysis. 
+However, we might be interested to see the expression of genes that are differentially expressed in other cell types present in our samples.
 
-```{r}
-# let's look across some other cell types
-celltypes <- c("Tumor_Myoblast", "Tumor_Mesoderm", "Tumor_Myocyte", "Vascular Endothelium")
+```{r celltype comparison}
+# let's compare gene expression across some other cell types
+# look at all tumor cells and pick one normal cell type
+celltypes <- c("Tumor_Myoblast", 
+               "Tumor_Mesoderm", 
+               "Tumor_Myocyte", 
+               "Vascular Endothelium")
 
 # subset to just tumor celltypes that we are interested in
 tumor_sce <- rms_sce[, which(rms_sce$celltype_broad %in% celltypes)]
 ```
 
+Next we will look at a few DE genes that we identified, one up regulated gene and one down regulated gene, and compare their expression in myoblasts to other cell types in ARMS and ERMS samples.
+We will use the `scater::plotExpression()` function to create a violin plot with RMS subtype on the x-axis and gene expression on the y-axis.
+We can continue using `facet_grid()` to show separate panels for each cell type.
+Because we want to show multiple genes here, we are going to add an additional option to `facet_grid()` to include multiple rows in our plot grid, one for each gene of interest.
 
-```{r}
-# pick a few genes to look at 
+```{r multi-gene plot}
+# pick a couple genes to look at 
 genes_to_plot <- c("ENSG00000196090", #PTPRT
                    "ENSG00000148935") #GAS2
 
-# create a violin plot across cell types of interest
+# create a violin plot 
 scater::plotExpression(tumor_sce,
-                       features = genes_to_plot, #PTPRT
+                       features = genes_to_plot, # a vector of genes to plot
                        x = "diagnosis_group", 
                        colour_by = "diagnosis_group",
                        other_fields = "celltype_broad",
                        point_size = 0.1) +
-  facet_grid(Feature ~ celltype_broad) + 
+  facet_grid(cols = vars(celltype_broad), # each celltype is its own column
+             rows = vars(Feature)) + # each feature is its own row
```

Comment:
`scater::plotExpression()` actually creates this `Feature` column. We should explain this! Also this is why it doesn't have to be in the `other_fields` list.